aea/skills/base.py

```diff
@@ -323,6 +330,9 @@ def teardown(self) -> None:
 
         :return: None
         """
+        super_obj = super()
+        if hasattr(super_obj, "teardown"):
+            super_obj.teardown()  # type: ignore  # pylint: disable=no-member
```

Comment:
why are we adding this here in the abstract methods?

Reply:
I'm not sure it's not the most elegant way with this hasattr/super, but every place it was added it's required.

The main problem is multiple inheritances, so it can be resolved in various ways. depending on base classes order.